File: solutions/java/triangle/1/src/main/java/Triangle.java
class Triangle {
    private double a, b, c;

    
    Triangle(double side1, double side2, double side3) throws TriangleException {
        if (
            side1 < side2 + side3 && side2 < side1 + side3 &&
            side3 < side1 + side2
            ) {
            
            this.a = side1;
            this.b = side2;
            this.c = side3;
        } else {
            throw new TriangleException();
        }
    }

    boolean isEquilateral() {
        return this.a == this.b && this.b == this.c;
    }

    boolean isIsosceles() {
        return this.a == this.b || this.b == this.c || this.c == this.a;
    }

    boolean isScalene() {
        return !isIsosceles();
    }

}

File: solutions/c/triangle/1/src/triangle.c
#include "triangle.h"

bool is_triangle(triangle_t t) {
  return t.a + t.b > t.c && t.a + t.c > t.b && t.b + t.c > t.a;
}

bool is_isosceles(triangle_t t) {
  return is_triangle(t) && (t.a == t.b || t.a == t.c || t.b == t.c);
}

bool is_equilateral(triangle_t t) {
  bool result = t.a == t.b && t.a == t.c;
  if (t.a == 0) return false;
  return result;
}

bool is_scalene(triangle_t t) {
  return is_triangle(t) && !is_isosceles(t) && !is_equilateral(t);
}